packages/core/src/util/LowPrecisionTaskScheduler/LowPrecisionTaskScheduler.ts

```diff
@@ -52,11 +52,11 @@ const addTask = ({key, firingDate, task, intervalDelay}: ScheduleLowPrecisionTas
     const tasks = intervals[intervalDelay]?.tasks;
     if (tasks?.length !== 0) {
       for (const taskData of tasks) {
-        if (nowTime >= firingDate) {
+        if (nowTime >= taskData.firingDate) {
           const {task, key} = taskData;
           logger.info(`Executing task with key "${key}"`);
-          task();
           cancelTask({intervalDelay, key});
+          task();
```

Comment:
Moved execution of task after canceling the current one in case where the task itself contained the logic of re-scheduling itself.